geniza/entities/models.py

```diff
@@ -146,6 +151,168 @@ def __str__(self):
+                person_description = getattr(person, description_field)
+                current_description = getattr(self, description_field)
+                if person_description and person_description not in current_description:
+                    description_chunks[lang_code].append(
+                        "Description from merged entry:\n%s" % (person_description,)
+                    )
+
+            # combine person-person relationships
+            for to_relationship in person.to_person.all():
+                # start with "to" relations; merged person is "from_person"
+                # don't add relationships where the other person is self!
+                if to_relationship.to_person.pk != self.pk:
+                    # set self to the side that was the merged person
+                    to_relationship.from_person = self
+                    to_relationship.save()
+            for from_relationship in person.from_person.all():
+                # repeat for "from" relations; merged person is "to_person"
+                if from_relationship.from_person.pk != self.pk:
+                    from_relationship.to_person = self
+                    from_relationship.save()
+
+            # combine person-document relationhips
+            for doc_relationship in person.persondocumentrelation_set.all():
+                # prevent duplicates (by document and relation type)
+                if not self.persondocumentrelation_set.filter(
+                    document=doc_relationship.document,
+                    type=doc_relationship.type,
+                ).exists():
+                    # reassign to self
+                    doc_relationship.person = self
+                    doc_relationship.save()
+
+            # combine person-place relationhips
+            for place_relationship in person.personplacerelation_set.all():
+                # prevent duplicates (by place and relation type)
+                if not self.personplacerelation_set.filter(
+                    place=place_relationship.place,
+                    type=place_relationship.type,
+                ).exists():
+                    # reassign to self
+                    place_relationship.person = self
+                    place_relationship.save()
+
+            # combine footnotes
+            for footnote in person.footnotes.all():
+                if not self.footnotes.includes_footnote(footnote):
+                    # if there is not a matching one on this person, can add footnote
+
+                    # first remove any doc relations; person footnotes should not have
+                    # doc_relation anyway, but just in case...
+                    if footnote.doc_relation:
+                        footnote.doc_relation = ""
+                        footnote.save()
+
+                    # then add to this person
+                    self.footnotes.add(footnote)
+
+        # combine aggregated content for text fields
```

Comment:
wondered why this was so far from where it's collected, but I see now there's a change in indentation - so this is after content has been aggregated from all records being merged?

Reply:
Right, all the description chunks are collected for each merged person, and then combined at the end of the loop. This is also from the document merge verbatim.